src/Textarea.jsx

```diff
@@ -282,6 +298,11 @@ class ReactTextareaAutocomplete extends React.Component<Props, State> {
       onChange(e);
     }
 
+    if (onCaretPositionChange) {
+      const caretPosition = this.getCaretPosition();
+      onCaretPositionChange(caretPosition);
+    }
+
```

Comment:
@jukben responding to your comment above - here is the change handler call after each "keypress" you expected. Whenever the value of the textarea is change the caret position is updated and the handler is called.

Reply:
@marcinlichwala Oh, I see. You are correct, but I think that this doesn't cover the cases when the caret is changed by mouse click, am I right?

Reply:
You are definitely right 👍 will need to improve that, good call!

Reply:
I might use the onSelect handler on the textarea itself: https://codesandbox.io/s/p9zmr1yqj0

Reply:
Yeah, good idea! `onSelect` seems like the correct place for it. You could remove it from `_changeHandler` and place it only there. 👍